### PR TITLE
Mjbots behavior loop and example

### PR DESCRIPTION
**Pull request by ethanmusser**
A class extending `MjbotsControlLoop` to compose a `kodlab::BehaviorManager` object and wrap some of the `BehaviorManager`'s functions.  This is intended to be extended by the user and act as the main control loop.

- Provides an `MjbotsBehaviorLoop` class in the `kodlab::mjbots` namespace that holds a `kodlab::BehaviorManager` object.
  - This object extends `MjbotsControlLoop` and maintains the same template and intended implementation.
  - The `MjbotsControlLoop` `Update` method is overridden to call the `BehaviorManager` `Update`, which itself calls the `Update`'s of the selected behavior (as well as handling their `Begin` and `Stop` methods).
  - Provides wrappers for the `AddBehavior` and `SetDefaultbehavior` functions of the composed `BehaviorManager` which automatically include the shared pointer to the `RobotBase`-derived robot.
- Provides an example of a user-implemented behavior loop which maintains several custom behaviors and demonstrates switching between them via LCM input.

This implementation has been tested with the `behavior_robot_example.cpp` example provided, which uses the various example behaviors provided in #39.

Note that this PR is 3/3 in the behavior implementation series and depends on the `Behavior` class introduced in #39 and the `BehaviorManager` class introduced in #40.

### CODE REVIEW

File: examples/behavior_robot_example.cpp [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+/**
+ * @brief Main method
+ * @return 0 if program exited without issue
+ */
+int main(int argc, char **argv)
+{
+  // Setup joints with a std::vector
+  std::vector<kodlab::mjbots::JointMoteus> joints;
+  joints.emplace_back(22, 1, 1, 0, 1.0, 5.0);
+//  joints.emplace_back(101, 4, -1, 0, 5.0 / 3.0, 0);
```
Comment (jdcaporale) [inline examples/behavior_robot_example.cpp:85]: remove this probably

File: examples/behavior_robot_example.cpp [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+  // Define robot options
+  kodlab::mjbots::ControlLoopOptions options;
+  options.log_channel_name = "motor_data";  // LCM topic for logging
+  options.input_channel_name =
+      "mode_input";  // LCM topic for input (must match broadcasting topic)
+  options.frequency = 1000;  // control loop frequency
+  options.realtime_params.main_cpu = 3;
+  options.realtime_params.can_cpu = 2;
+  options.max_torque = 20;
```
Comment (jdcaporale) [inline examples/behavior_robot_example.cpp:95]: Probably lower this significantly

File: include/kodlab_mjbots_sdk/mjbots_behavior_loop.h [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+ */
+template<class Log = VoidLcm, class Input = VoidLcm,
+    class Robot = kodlab::RobotBase>
+class MjbotsBehaviorLoop : public MjbotsControlLoop<Log, Input, Robot>
+{
+
+protected:
+  /**
+   * @brief Updates robot state and sets torques based on active behavior in
+   *        behavior manager
```
Comment (jdcaporale) [inline include/kodlab_mjbots_sdk/mjbots_behavior_loop.h:41]: @note If a derived class chooses to override update(), ensure that the necessary internal updates are called. (robot_ and behavior_mgr

File: include/kodlab_mjbots_sdk/mjbots_behavior_loop.h [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+  /**
+   * @brief Destructor
+   */
+  virtual ~MjbotsBehaviorLoop() {}
+
+  /**
+   * @brief Behavior manager handling robot behaviors and transitions between
+   *        robot behaviors
+   */
+  BehaviorManager<Robot> behavior_mgr;
```
Comment (jdcaporale) [inline include/kodlab_mjbots_sdk/mjbots_behavior_loop.h:92]: Should this be private?

File: include/kodlab_mjbots_sdk/type_traits.h [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+ * @brief Static test to determine if class template `Ts` is a base of class `C`
+ * @warning This expression will not work in cases of private inheritance, and
+ * may not work in cases of multiple inheritance.
+ * @cite  https://stackoverflow.com/a/34672753
+ * @tparam C concrete class based on a class template
+ * @tparam Ts base class template
+ * @return std::true_type if `Ts` is a base class template for `C`,
+ * std::false_type otherwise
+ */
+template<typename Ts, template<typename...> class C>
```
Comment (jdcaporale) [inline include/kodlab_mjbots_sdk/type_traits.h:52]: Ts is a bad name I think, should just be word like Derived or something

Comment (jdcaporale) [reply include/kodlab_mjbots_sdk/type_traits.h:52]: Really BaseClassTemplate

Comment (ethanmusser) [reply include/kodlab_mjbots_sdk/type_traits.h]: Fixed in 9598bd26f2f7fa257d51027057928d285d5549b2

File: examples/behavior_robot_example.cpp (hunk 1 above)
Comment (ethanmusser) [reply examples/behavior_robot_example.cpp]: Fixed in 8cecc842b6606aa38becbff419a2a1fc14f11dbb.

File: examples/behavior_robot_example.cpp (hunk 2 above)
Comment (ethanmusser) [reply examples/behavior_robot_example.cpp]: Fixed in 8cecc842b6606aa38becbff419a2a1fc14f11dbb.

File: include/kodlab_mjbots_sdk/mjbots_behavior_loop.h (hunk 3 above)
Comment (ethanmusser) [reply include/kodlab_mjbots_sdk/mjbots_behavior_loop.h:45]: Fixed in 8cecc842b6606aa38becbff419a2a1fc14f11dbb.

File: include/kodlab_mjbots_sdk/mjbots_behavior_loop.h (hunk 4 above)
Comment (ethanmusser) [reply include/kodlab_mjbots_sdk/mjbots_behavior_loop.h]: Changed in 8cecc842b6606aa38becbff419a2a1fc14f11dbb.

**Review by ShaneRozenLevy — APPROVED** [review]
LGTM

File: README.md [hunk 6]
```diff
@@ -146,6 +146,14 @@ the vector and the default behavior can be set by the user. The
 and used to maintain a series of behaviors running on a`RobotBase`-derived 
 robot. 
 
+## Mjbots Behavior Loop
+The `MjbotsBehaviorLoop` extends the [`MjbotsControlLoop`](https://github.com/KodlabPenn/kodlab_mjbots_sdk#mjbotscontrolloop)
+to include a `BehaviorManager` which internally manages `Behavior` objects for a
+`RobotBase`-derived class.  The `MjbotsBehaviorLoop` works in much the same way
+as the `MjbotsControlLoop`, except the user no longer needs to override the
+`Update()` method.  An example demonstrating usage of the `MjbotsBehaviorLoop`
```
[the remark this answers is not in the document]
Comment (ethanmusser) [reply README.md]: Clarified in 04d8eb77efd5c78910a30913dd4ba4736b66f78e

**Pull request merged**